### PR TITLE
Switch from timestamp to sequence id for avatar identity

### PR DESCRIPTION
**Pull request by davidkelly**
DNM for now -- need to test this and having a PR makes that more convenient.

### CODE REVIEW

File: libraries/avatars/src/AvatarData.cpp [hunk 1]
```diff
@@ -1500,20 +1500,14 @@ QUrl AvatarData::cannonicalSkeletonModelURL(const QUrl& emptyURL) const {
 }
 
 void AvatarData::processAvatarIdentity(const Identity& identity, bool& identityChanged, bool& displayNameChanged, const qint64 clockSkew) {
```
Comment (howard-stearns) [inline libraries/avatars/src/AvatarData.cpp]: Do we need the clockSkew argument any more?

File: libraries/networking/src/udt/PacketHeaders.h [hunk 2]
```diff
@@ -234,7 +234,8 @@ enum class AvatarMixerPacketVersion : PacketVersion {
     VariableAvatarData,
     AvatarAsChildFixes,
     StickAndBallDefaultAvatar,
-    IdentityPacketsIncludeUpdateTime
+    IdentityPacketsIncludeUpdateTime,
```
Comment (howard-stearns) [inline libraries/networking/src/udt/PacketHeaders.h]: Don't we also have to change the reference to IdentityPacketsIncludeUpdateTime in versionForPacketType?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 689a0b5badb2479b382aee3d783e68939fb99b6f available at: https://gist.github.com/8d45f209fdcf80caac9b948de664e514<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10489-689a0b5.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10489-689a0b5.dmg

File: libraries/avatars/src/AvatarData.cpp (hunk 1 above)
Comment (davidkelly) [inline libraries/avatars/src/AvatarData.cpp]: oh crap -- good call

File: libraries/networking/src/udt/PacketHeaders.h (hunk 2 above)
Comment (davidkelly) [inline libraries/networking/src/udt/PacketHeaders.h]: No - but I forgot to update PacketHeaders.cpp to refer to it.  I'll do that now.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 057718bde3c6d8aff13cfff755d54c6ded3106c5 available at: https://gist.github.com/8d45f209fdcf80caac9b948de664e514<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10489-057718b.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10489-057718b.dmg

**Comment by howard-stearns** [comment]
This is item #1 in https://highfidelity.fogbugz.com/f/cases/4801/T-posers-and-unable-to-use-PAL

The test plan we followed was:
- confirmed that we could not connect anywhere else
- ran local server and connected. 
- saw each other and avatars, changed display name and avatar. shook hands. exited and returned.

**Comment by birarda** [comment]
Ideally would like to see us front-load the sequence number so that we can immediately not process an identity packet if we know we are just going to drop it on the floor.

Means we'd have to do that check in `AvatarData::parseAvatarIdentityPacket` I suppose.

**Comment by davidkelly** [comment]
At first glance, doing that check in `parseAvatarIdentityPacket`, (at least on the mixer side), saves very little.  All we do is call that function, set two bools and call `processAvatarIdentity`.  Not sure this saves much.  Perhaps there's something I'm missing?

**Comment by birarda** [comment]
I'm specifically hoping to avoid

```
packetStream >> identityOut.uuid
                 >> identityOut.skeletonModelURL
                 >> identityOut.attachmentData
                 >> identityOut.displayName
                 >> identityOut.sessionDisplayName
                 >> identityOut.avatarEntityData
```

when we don't need to do it

**Comment by hifi-gustavo** [comment]
Build console output for commit: 057718bde3c6d8aff13cfff755d54c6ded3106c5 available at: https://gist.github.com/8d45f209fdcf80caac9b948de664e514<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10489-057718b.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10489-057718b.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 95111d24b29f24b47f8c31154f49703d8a381fb3 available at: https://gist.github.com/8d45f209fdcf80caac9b948de664e514<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10489-95111d2.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10489-95111d2.dmg

**Pull request merged**